src/ray/core_worker/core_worker.h

```diff
@@ -27,7 +27,7 @@ class CoreWorker {
       const std::unordered_map<std::string, double> &required_resources,
       const std::vector<std::shared_ptr<RayObject>> &args,
       const std::vector<ObjectID> &arg_reference_ids,
-      const std::vector<ObjectID> &return_ids,
+      const std::vector<ObjectID> &return_ids, const bool return_results_directly,
```

Comment:
I agree we should also do this for Python as well, in fact is is probably necessary for supporting pickle5 objects as well.

Since this is a non-trivial amount of code to move from Python to C++ though, I filed a separate issue to do this:

Reply:
makes sense, thanks!